### DISCUSSION TITLE
Automation to update row column based on another column value

### DISCUSSION THREAD

**Discussion by tglynx** [Budibase/budibase#10723] [Help] [answered]
Hi,

my initial problem was that i want to sort a table using the values of a 'formula' field. I thought that this would be possible using a static formula as this seems to be the same as a text field, bit it is not, and this does not seem to work.
However, to work around this issue my idea was to migrate the column from a 'formula' typed column to a simple text column.
As it is not possible to change the datatype of a column from 'formula' to text' my idea was to create a new text column and migrate the content of the formula column to the newly created empty text column.

My approach to to this was to create an automation.
In step 1 i query rows from the table without filter.
In step 2  (update row) i create a loop of type array using {{ steps.1.rows }} as binding.
In step 3 (now update row) i want to simply update the column 'newtextcolumn' with the value of 'oldformulacolumn'.

For this purpose i assumed that i need to set the update row mode to 'use binding' and put something like {{ loop.currentItem.oldformulacolumn }} in the binding of newtextcolumn. 

This however, does not work, i get the following output: 

```
{
  "success": true,
  "items": [
    {
      "success": false,
      "response": {
        "message": "Invalid inputs"
      }
    }
  ],
  "iterations": 1
}
```

I assume i am doing it wrong, please advice.

Regards,
Michael

**Comment by ConorWebb96** [comment]
Hey @tglynx,

I actually answered a discussion I believe to be quite similar to this here; https://github.com/Budibase/budibase/discussions/10644, If I'm wrong and this doesn't help let me know and I'll take another look at this for you. You may not need the update automation if you're value doesn't constantly change.

Currently you can't check for changes of a specific row within an automations, however there is an issue for this here: https://github.com/Budibase/budibase/issues/10035, hopefully it will be added soon.

I hope this is helpful.

**Comment by tglynx** [comment]
Hey Conor, 

thanks, what i want to do is to get rid of the formula field i am currently using because the value does indeed not change constantly. I want to use a text field instead - so for existing records my plan was to use a automation to update a new textfield with the static value of the formula field initially. 

I am using a automation as stated above: step 1 query rows, step 2 loop with update rows.
I can confirm that {{ loop.currentItem.FirstName }} gives me the correct value i want to put into my new empty text field 'FirstNameStatic'.
I only have the columns configured with a binding above in the 'update row' step that need to be updated (that is correct i guess?).

However i get the error message above in the update row step (not understanding why).

**Reply by tglynx** [reply]
Well i got a little bit further, however, i am really confused now. 
I got the 'invalid inputs' issue resolved by putting {{ loop.currentItem._Id }} in the 'Row ID' field of the binding in the update row step.
However, a new row gets created with all columns empty, except the one i wanted to update ... i am obviously doing something wrong.

**Reply by tglynx** [reply, marked as answer]
again i got a little bit further realizing that it must be {{ loop.currentItem._id }} (lower case _id) ... 

The correct record gets updated now! Great :) 

All relationships are cleared when using 'update row'! However, i have not checked the 'clear relationships if empty' checkbox. 
I know -> completely different thing now, but this really confuses me!

**Comment by tglynx** [comment]
**Thanks Conor for pointing me into the right direction.** However, i am not sure if i found a issue with the 'update row' step!
When the binding is empty for a relationship and the 'clear relationships if empty' checkbox is not set, the relationships should not be touched when using 'update row' in a automation.
